### ISSUE TITLE
AnchorNode.anchor setter does not detach the replaced anchor → ARCore anchor leak

### ISSUE THREAD

**Issue by thomas-gorisse** [sceneview/sceneview#2043]
## Summary

`AnchorNode.anchor` is a public mutable property. Its setter swaps the backing field but never `detach()`es the previous `Anchor`. ARCore anchors carry ongoing per-frame processing cost — the node's own KDoc says so — yet replacing the anchor leaks the old one.

## Evidence

`arsceneview/src/main/java/io/github/sceneview/ar/node/AnchorNode.kt:48-54`:

```kotlin
var anchor: Anchor = anchor
    set(value) {
        field = value                      // old anchor is dropped, never detach()ed
        trackingState = value.trackingState
        pose = value.pose
        onAnchorChanged?.invoke(value)
    }
```

The class KDoc itself (`AnchorNode.kt:45-46`):

> *"Anchors incur ongoing processing overhead within ARCore. To release unneeded anchors use [destroy]."*

`destroy()` detaches the *current* anchor, but a programmatic re-assignment (`anchorNode.anchor = newAnchor`) strands the previous one — it stays attached to the `Session` and keeps accruing tracking cost until the whole session is destroyed.

The internal move-gesture path (`onMoveBegin` → `detachAnchor()`, `onMoveEnd` → `anchor = createAnchor()`) happens to be safe only because `onMoveBegin` detaches first. Any other caller setting `anchor` directly leaks.

## Why it matters

- ARCore documents a soft cap (~tens) of simultaneously-tracked anchors and warns each one costs CPU every frame. An app that re-anchors a node (re-localization, snapping to a new plane, cloud-anchor resolve replacing a local anchor) leaks one ARCore anchor per reassignment.
- Cross-check: `CloudAnchorNode.resolve` creates a brand-new `CloudAnchorNode` rather than reusing one, partly sidestepping this — but the public setter remains a footgun.

## Acceptance criteria

- The `anchor` setter detaches the previous anchor before replacing it (e.g. `field.takeIf { it != value }?.detach()` — guard against detaching an anchor still in use, and against detaching the same instance).
- Decide and document whether the setter owns the lifecycle of the assigned anchor; if a caller may want to keep the old anchor, provide an opt-out or document that they must retain it.
- A unit/instrumented test asserts that reassigning `anchor` detaches the old one exactly once.
